src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/entity/Assessment.kt

```diff
@@ -29,8 +33,12 @@ data class Assessment(
   val offender: Offender,
 
   @NotNull
-  @Enumerated(EnumType.STRING)
-  var status: AssessmentStatus = AssessmentStatus.NOT_STARTED,
+  @Embedded
+  @AttributeOverrides(
+    AttributeOverride(name = "status", column = Column(name = "status")),
+    AttributeOverride(name = "previousStatus", column = Column(name = "previous_status")),
+  )
+  var status: AssessmentState = AssessmentState.NotStarted,
```

Comment:
the idea would be rather than storing a enum against the record we'd stored an embeddable State object - which consist of an enum and possibly extra fields that are required to reconstitute the state.

So like if someone opts out, the opt out state also includes info about where it was opted out from (previous stage) - likely to need other things for time out

It'd be difficult to support when you have multiple levels of states you need to return from:
 ELIGIBLE -> OPT OUT -> TIMED OUT -> OPT OUT -> ELIGIBLE 

We have a history of assessment events we could possibly look at using.. 


These would be columns on the Assessment table

Reply:
this isn't currently working - relies on persisting instances of a sealed class as an embedded entity.